dlt/destinations/dataset/relation.py

```diff
@@ -213,72 +243,150 @@ def __init__(
         limit: int = None,
         selected_columns: Sequence[str] = None,
         normalize_query: bool = True,
+        sqlglot_expression: sge.Select = None,
     ) -> None:
-        """Create a lazy evaluated relation to for the dataset of a destination"""
+        """Create a lazy evaluated relation for the dataset of a destination"""
 
         # NOTE: we can keep an assertion here, this class will not be created by the user
-        assert bool(table_name) != bool(
-            provided_query
-        ), "Please provide either an sql query OR a table_name"
-
+        assert (
+            sum(x is not None for x in (provided_query, table_name, sqlglot_expression)) == 1
+        ), "Please provide either an sql query, a table_name or a an explicit sqlglot expression"
         super().__init__(readable_dataset=readable_dataset, normalize_query=normalize_query)
 
         self._provided_query = provided_query
-        self._table_name = table_name
-        self._limit = limit
         self._selected_columns = selected_columns
 
-    def _query(self) -> Any:
-        # TODO reimplement this using SQLGLot instead of passing strings
-        if self._provided_query:
-            return self._provided_query
-
-        dataset_schema = self._dataset.schema
-
-        table_name = (
-            self._table_name
-        )  # self.sql_client.make_qualified_table_name(self._table_name, quote=False, casefold=False)
-
-        maybe_limit_clause_1 = ""
-        maybe_limit_clause_2 = ""
-        if self._limit:
-            maybe_limit_clause_1, maybe_limit_clause_2 = self.sql_client._limit_clause_sql(
-                self._limit
+        self._sqlglot_expression: sge.Select = None
+        if sqlglot_expression:
+            self._sqlglot_expression = sqlglot_expression
+        elif provided_query:
+            self._sqlglot_expression = cast(
+                sge.Select,
+                parse_one(
```

Comment:
you can do `maybe_parse` here, and it will either take the expression if it is one, or parse a string.